internal/osutil/mount_linux.go

```diff
@@ -30,3 +43,30 @@ func IsMounted(baseDir string) (bool, error) {
 	}
 	return false, nil
 }
+
+// Mount attaches a filesystem accessible via the device node specified by source
+// to the specified baseDir. If not existing, baseDir will be created before
+// mounting the filesystem.
+func Mount(source, baseDir, fstype string, readOnly bool) error {
+	if err := os.MkdirAll(baseDir, 0755); err != nil {
+		return fmt.Errorf("cannot create directory %q: %w", baseDir, err)
+	}
+
+	flags := uintptr(0)
+	if readOnly {
+		flags |= unix.MS_RDONLY
+	}
+	if err := syscallMount(source, baseDir, fstype, flags, ""); err != nil {
+		return fmt.Errorf("cannot mount %q: %w", source, err)
```

Comment:
Mounting is the main purpose of the function, so no need to wrap the error with this message, otherwise the caller will probably duplicate this in the error message. Similar with `syscallUnmount` below.